packages/flutter/lib/src/widgets/splash.dart

```diff
@@ -0,0 +1,621 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'lookup_boundary.dart';
+import 'notification_listener.dart';
+import 'ticker_provider.dart';
+
+// Examples can assume:
+// late Color color;
+// late BuildContext context;
+
+/// Signature for the callback used by [Splash]es to obtain the
+/// appropriate [Rect].
+typedef RectCallback = Rect Function();
+
+/// A splash of [color] shown in response to a user gesture.
+///
+/// The [confirm] and [cancel] methods alter the splash based on the
+/// user's behavior. For example, a tap event might trigger a splash
+/// that's confirmed when the corresponding [GestureDetector.onTapUp]
+/// event is seen and canceled when the pointer is dragged out of the
+/// reference box.
+///
+/// See also:
+///
+/// * [SplashBox], the widget that paints splashes.
+/// * [SplashFactory], a class that creates splashes using a specific
+///   function signature.
+/// * [InkResponse], which creates splashes using a [SplashFactory].
+abstract class Splash {
+  /// Creates an interactive splash effect.
```

Comment:
```suggestion
  /// Creates a splash effect.
```

Now that we've dropped the separation of the interactive and non-interactive classes the "interactive" word might not be worth mentioning.